pkg/shared/hacks/temp.go

```diff
@@ -0,0 +1,56 @@
+package hacks
+
+// Temporary hack package
+// Nothing to see here
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+
+	kafkamgmtclient "github.com/redhat-developer/app-services-sdk-go/kafkamgmt/apiv1/client"
+)
+
+// Temporary hack that we use to determine if
+// Our CLI needs to use mas-sso token
+func ShouldUseMasSSO() bool {
+	req, err := http.NewRequest("GET", "https://api.openshift.com/api/kafkas_mgmt/v1/sso_providers", nil)
+	if err != nil {
+		return true
+	}
+
+	req = req.WithContext(context.Background())
+
+	req.Header.Set("Accept", "application/json")
+
+	resp, err := http.DefaultClient.Do(req)
+	if err != nil {
+		return true
+	}
+
+	defer resp.Body.Close()
+
+	b, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return true
+	}
+
+	response := string(b)
+
+	// defining a struct instance
+	var provider *kafkamgmtclient.SsoProvider
+
+	responseBytes := []byte(fmt.Sprintf("%v", response))
+	err = json.Unmarshal(responseBytes, &provider)
+	if err != nil {
+		return true
+	}
+
+	if provider.GetBaseUrl() == "" {
+		return true
+	}
+
+	return provider.GetBaseUrl() == "https://identity.api.redhat.com"
```

Comment:
Is the `https://identity.api.redhat.com` the same for both environments? 
@wtrocki @akoserwal

Reply:
Amazing question. I do not know.

Reply:
I would love to have some flag on that API. Like useLegacyAuth: true. That will be much easier for us to pick up dynamic behaviour rather than assert urls

Reply:
SsoProvider: MAS SSO/RED Hat SSO  (Enum) could it be used?

Reply:
> I would love to have some flag on that API. Like useLegacyAuth: true. While it might look like pouting API, but it would in fact make things That will be much easier for us to pick up dynamic behaviour rather than assert urls
> 
> My ideas for that flag:
> 
> ```
>   useLegacyAuth: true
>   usePluralAuth: true
>   multiSSOLogin: true
>   customSSO: true
> ```

That's a very good suggestion and one for @Rajagopalan-Ranganathan and @akoserwal :-). 

> SsoProvider: MAS SSO/RED Hat SSO (Enum) could it be used?

I think so @Rajagopalan-Ranganathan

Reply:
Enums in API should be used for values that generally do not change: Like Months etc.
Using enum in the API that we planning to extend (or enums that doesn't have specified clear bounds would mean that API will be always breaking every time we add enum.

Reply:
For this change I think it is fine because the values are not going to change e.g a rename or a removal of one etc
Extending an enum is fine and should never ever be a breaking change IMO.

Reply:
Adding a bool flag-like    `useLegacyAuth: true` would be confusing to other clients that are consuming this end-point. I am planning to add a `name: mas_sso` or `name: redhat_sso`. 

@wtrocki @Rajagopalan-Ranganathan @carlesarnal @machi1990 wdty?

Reply:
I do not have strong opinion. I need value for `if statement` that we can rely on in CLI and UI and this proposal meets it.

Reply:
About the name makes sense. I don't have a strong opinion about the flag, but I would like to use whatever value is returned by the sso_providers call.

Reply:
the name is absolutely fine and looks to be extensible in the future.